auth.py

```
@@ -105,4 +105,100 @@ def complex_auth_function(user, pass_word, email, phone, address, city, state, z
+                    if user_data.get('phone'):
+                        if len(user_data['phone']) > 10:
+                            if user_data.get('address'):
+                                if user_data['address'].get('city'):
+                                    if user_data['address'].get('state'):
+                                        return True
+    return False
+
+# Issue 26: Performance - Inefficient list comprehension
+def filter_active_users(all_users):
+    active_users = [user for user in all_users if user['status'] == 'active']
+    return [user for user in active_users if user['last_login'] != None]
```

Ruff (0.12.2)

Comparison to `None` should be `cond is not None`

Replace with `cond is not None`

(E711)